File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/gettasktimephaseddata.rb
module Asposetasksjava
  module GetTaskTimephasedData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        prj = Rjb::import('com.aspose.tasks.Prj')
        tsk = Rjb::import('com.aspose.tasks.Tsk')

        cal = Rjb::import('java.util.Calendar').getInstance()
		cal.set(2013,7,17,8,0,0)
		project.set(prj.START_DATE, cal.getTime())
		project.set(prj.NEW_TASKS_ARE_MANUAL, Rjb::import('com.aspose.tasks.NullableBool').new(false))

		task = project.getRootTask().getChildren().add("Task")
		rsc = project.getResources().add("Rsc")
		rsc.set(Rjb::import('com.aspose.tasks.Rsc').STANDARD_RATE, Rjb::import('java.math.BigDecimal').valueOf(10))
		rsc.set(Rjb::import('com.aspose.tasks.Rsc').OVERTIME_RATE, Rjb::import('java.math.BigDecimal').valueOf(15))
		# 6 days duration
		task.set(tsk.DURATION, project.getDuration(6))

		assn = project.getResourceAssignments().add(task, rsc)
		d = Rjb::import('java.util.Date').new(0)
		assn.set(Rjb::import('com.aspose.tasks.Asn').STOP, Rjb::import('java.util.Date').new(0))
		assn.set(Rjb::import('com.aspose.tasks.Asn').RESUME, Rjb::import('java.util.Date').new(0))

		# backloaded contour increases task duration from 6 to 10 days
		assn.set(Rjb::import('com.aspose.tasks.Asn').WORK_CONTOUR, Rjb::import('com.aspose.tasks.WorkContourType').BackLoaded)

		project.setBaseline(Rjb::import('com.aspose.tasks.BaselineType').Baseline)

		task.set(tsk.PERCENT_COMPLETE, 50)

		td = assn.getTimephasedData(assn.get(Rjb::import('com.aspose.tasks.Asn').START), 
									assn.get(Rjb::import('com.aspose.tasks.Asn').FINISH), 
									Rjb::import('com.aspose.tasks.TimephasedDataType').AssignmentRemainingWork).toList()

		puts "Size: " + td.size().to_s
		puts "Value: " + td.get(0).getValue().to_s
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/GetTaskTimephasedData.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Prj as Prj;
use com\aspose\tasks\Tsk as Tsk;
use com\aspose\tasks\NullableBool as NullableBool;
use com\aspose\tasks\Rsc as Rsc;
use com\aspose\tasks\Asn as Asn;
use com\aspose\tasks\WorkContourType as WorkContourType;
use com\aspose\tasks\BaselineType as BaselineType;
use com\aspose\tasks\TimephasedDataType as TimephasedDataType;

use java\util\Calendar as Calendar;
use java\math\BigDecimal as BigDecimal;
use java\util\Date as Date;
class GetTaskTimephasedData{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $prj = new Prj();
        $tsk = new Tsk();

        $cal=new Calendar();
        $cal = $cal->getInstance();

		$cal->set(2013,7,17,8,0,0);
		$project->set($prj->START_DATE, $cal->getTime());
		$project->set($prj->NEW_TASKS_ARE_MANUAL, new NullableBool(false));

		$task = $project->getRootTask()->getChildren()->add("Task");
		$rsc = $project->getResources()->add("Rsc");

        $rscc=new Rsc();
        $bigDecimal=new BigDecimal();

		$rsc->set($rscc->STANDARD_RATE, $bigDecimal->valueOf(10));
		$rsc->set($rscc->OVERTIME_RATE, $bigDecimal->valueOf(15));
		# 6 days duration
		$task->set($tsk->DURATION, $project->getDuration(6));

		$assn = $project->getResourceAssignments()->add($task, $rsc);
		$d = new Date(0);
        $asnn=new Asn();

        $assn->set($asnn->STOP, new Date(0));
		$assn->set($asnn->RESUME, new Date(0));

		# backloaded contour increases task duration from 6 to 10 days
        $workContourType=new WorkContourType();

		$assn->set($asnn->WORK_CONTOUR, $workContourType->BackLoaded);

        $baselineType=new BaselineType();
		$project->setBaseline($baselineType->Baseline);

		$task->set($tsk->PERCENT_COMPLETE, 50);

        $timephasedDataType=new TimephasedDataType();

		$td = $assn->getTimephasedData($assn->get($asnn->START),
                $assn->get($asnn->FINISH),
                $timephasedDataType->AssignmentRemainingWork)->toList();

		print "Size: " . (string)$td->size().PHP_EOL;//.to_s
		print "Value: " . (string)$td->get(0)->getValue().PHP_EOL;//.to_s
    }

}
?>